internal/stack/serverlessresources_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stack
+
+import (
+	"testing"
+)
+
+func TestEsHostWithPort(t *testing.T) {
+	tests := []struct {
+		name string
+		host string
+		want string
+	}{
+		{"host without port", "https://hostname", "https://hostname:443"},
+		{"host with port", "https://hostname:443", "https://hostname:443"},
+		{"host with differernt port", "https://hostname:9200", "https://hostname:9200"},
+		{"ipv6 host", "http://[2001:db8:1f70::999:de8:7648:6e8]:100/", "http://[2001:db8:1f70::999:de8:7648:6e8]:100/"},
+		{"ipv6 host without port", "http://[2001:db8:1f70::999:de8:7648:6e8]", "http://[2001:db8:1f70::999:de8:7648:6e8]:443"},
+	}
```

Comment:
Not sure if we can have ES hosts with path, but we can with Kibana and maybe other hosts. I think we should cover this case just in case this can happen, or if we reuse this code in the future and forget about this case.
```suggestion
		{"host with path", "https://hostname/xyz", "https://hostname:443/xyz"},
		{"ipv6 host with path", "https://[::1]/xyz", "https://[::1]:443/xyz"},
	}
```